src/main/java/slimeknights/tconstruct/library/recipe/casting/ICastingRecipe.java

```diff
@@ -0,0 +1,74 @@
+package slimeknights.tconstruct.library.recipe.casting;
+
+import net.minecraft.fluid.Fluid;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraftforge.fluids.FluidStack;
+import slimeknights.tconstruct.library.recipe.RecipeTypes;
+import slimeknights.tconstruct.smeltery.TinkerSmeltery;
+import slimeknights.tconstruct.smeltery.recipe.ICastingInventory;
+
+public interface ICastingRecipe extends IRecipe<ICastingInventory> {
+  @Override
+  default ItemStack getCraftingResult(ICastingInventory inv) {
+    return getRecipeOutput().copy();
+  }
+
+  @Override
+  default boolean canFit(int width, int height) {
+    return true;
+  }
+
+  @Override
+  default boolean isDynamic() {
+    return true;
+  }
+
+  @Override
+  default ItemStack getIcon() {
+    return new ItemStack(getType() == RecipeTypes.CASTING_TABLE ? TinkerSmeltery.castingTable : TinkerSmeltery.castingBasin);
+  }
+
+  /**
+   * Gets the amount of fluid required for this recipe
+   * @param inv  Inventory instance
+   * @return  Fluid amount when using the fluid in the inventory
+   */
+  int getFluidAmount(ICastingInventory inv);
+
+  boolean isConsumed();
```

Comment:
Java docs on the unique methods is useful. This one says something like `@returns true if the cast item is consumed when crafting`